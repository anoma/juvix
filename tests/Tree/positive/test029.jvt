type list {
  nil : list;
  cons : (*, list) â†’ list;
}

function main() : *;

function main() : * {
  seq(seq(seq(seq(seq(seq(seq(save(eq(1, 1)) {
    seq(trace(tmp[0]), tmp[0])
  }, save(eq(0, 1)) {
    seq(trace(tmp[0]), tmp[0])
  }), save(eq(alloc[nil](), alloc[nil]())) {
    seq(trace(tmp[0]), tmp[0])
  }), save(eq(alloc[cons](1, alloc[nil]()), alloc[nil]())) {
    seq(trace(tmp[0]), tmp[0])
  }), save(eq(alloc[cons](1, alloc[nil]()), alloc[cons](2, alloc[nil]()))) {
    seq(trace(tmp[0]), tmp[0])
  }), save(eq(alloc[cons](1, alloc[nil]()), alloc[cons](1, alloc[cons](2, alloc[nil]())))) {
    seq(trace(tmp[0]), tmp[0])
  }), save(eq(alloc[cons](1, alloc[cons](2, alloc[nil]())), alloc[cons](1, alloc[cons](2, alloc[nil]())))) {
    seq(trace(tmp[0]), tmp[0])
  }), void)
}
