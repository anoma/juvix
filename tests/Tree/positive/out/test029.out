true
false
true
false
false
false
true
